src/preload/client/files.ts

```diff
@@ -88,4 +88,23 @@ export class FilesClient {
       });
     });
   };
+
+  uploadFile = async (file: File): Promise<UploadResponse> => {
```

Comment:
Worth comparing to the `upload = async (dataUrl: string): Promise<UploadResponse>` implementation above; I had to implement the above one like that because Plate's `createImagePlugin` reads the dropped file as a dataUrl before handing off. 

By comparison, (as I learned) you can directly hand off an `ArrayBuffer` (I believe) from the dropped file and stream it to the filesystem. This saves the need to read the file into memory again as a dataUrl, and then to encode it from dataUrl back to binary. Probably not super noticeable performance wise, but its notable because the implementation is actually simpler in addition to being more performant; I think generic file handling could probably merge all the plugins together and go through this method.